tests/test-team/template-mgmt-component-tests/template-mgmt-choose-page.component.ts

```diff
@@ -130,42 +167,106 @@ test.describe('Choose Template Type Page', () => {
+        sessionDataForRadioSelect.id
+      );
+      await TemplateMgmtChoosePage.checkRadioButton(page.getByLabel(label));
+      await chooseTemplatePage.clickContinueButton();
+
+      await expect(page).toHaveURL(
+        `${baseURL}/templates/create-${path}-template/${sessionDataForRadioSelect.id}`
+      );
+    });
+
+  for (const { label, sessionData } of [
+    { label: 'NHS App message', sessionData: nhsAppRadioSelectedSessionData },
+    { label: 'Email', sessionData: emailRadioSelectedSessionData },
+    { label: 'Text message (SMS)', sessionData: smsRadioSelectedSessionData },
+    { label: 'Letter', sessionData: letterRadioSelectedSessionData },
+  ]) {
+    test(`should display ${label} radio button selected if present in session storage`, async ({
+      page,
+    }) => {
+      const chooseTemplatePage = new TemplateMgmtChoosePage(page);
+
+      await chooseTemplatePage.navigateToChooseTemplatePage(sessionData.id);
+
+      expect(page.getByLabel(label)).toBeChecked();
+    });
+  }
+
+  test('should not display "Go Back" link on page', async ({ page }) => {
     const chooseTemplatePage = new TemplateMgmtChoosePage(page);
 
-    await chooseTemplatePage.navigateToChooseTemplatePage(
-      emptySessionDataForRadioSelect.id
-    );
-    await TemplateMgmtChoosePage.checkRadioButton(
-      page.locator('[id="templateType-NHS_APP"]')
-    );
-    await chooseTemplatePage.clickContinueButton();
+    await chooseTemplatePage.navigateToChooseTemplatePage(emptySessionData.id);
 
-    await expect(page).toHaveURL(
-      `${baseURL}/templates/create-nhs-app-template/${emptySessionDataForRadioSelect.id}`
-    );
+    await expect(chooseTemplatePage.goBackLink).toBeHidden();
   });
 
-  test('should display NHS App radio button selected if present in session storage', async ({
+  test('should navigate to error page if sessionId invalid', async ({
     page,
+    baseURL,
   }) => {
     const chooseTemplatePage = new TemplateMgmtChoosePage(page);
 
-    await chooseTemplatePage.navigateToChooseTemplatePage(
-      nhsAppRadioSelectedSessionData.id
-    );
+    await chooseTemplatePage.navigateToChooseTemplatePage('invalid-sessionId');
 
-    // expect to assert radio selected
+    await expect(page).toHaveURL(`${baseURL}/templates/invalid-session`);
   });
 
-  test('should not display "Go Back" link on page', async ({ page }) => {
+  test('Footer links exist and are visible', async ({ page }) => {
     const chooseTemplatePage = new TemplateMgmtChoosePage(page);
-
     await chooseTemplatePage.navigateToChooseTemplatePage(emptySessionData.id);
 
-    await expect(chooseTemplatePage.goBackLink).toBeHidden();
+    const footerLinks = [
+      {
+        name: 'Accessibility statement',
+        selector: 'a[data-testid="accessibility-statement-link"]',
+        href: '/accessibility',
+      },
+      {
+        name: 'Contact Us',
+        selector: 'a[data-testid="contact-us-link"]',
```

Comment:
can we avoid using data-testid? data-testid only exists for the benefit of the tests so you're not actually testing the page behaviour - do these links not have any other attributes?

Reply:
agree - copy and paste and updating

Reply:
updated to use role based locators as suggested by playwright